### PR TITLE
send notification to hosts, cohosts when someone joins the event wait…

### PR DESCRIPTION
**Pull request by superryeti**
Context: Follow up PR to https://github.com/MIT-LCP/physionet-buildtps://github.com/MIT-LCP/physionet-build/pull/1865

This PR adds the logic to send email notification to event host when someone joins the event.

Email for reference

```
physionet-build-dev-1   | Content-Type: text/plain; charset="utf-8"
physionet-build-dev-1   | MIME-Version: 1.0
physionet-build-dev-1   | Content-Transfer-Encoding: 7bit
physionet-build-dev-1   | Subject: DataShare New Event Registration
physionet-build-dev-1   | From: localhost@localhost
physionet-build-dev-1   | To: aewj@mit.edu
physionet-build-dev-1   | Date: Sat, 18 Feb 2023 03:52:23 -0000
physionet-build-dev-1   | Message-ID: <167669234328.52.17528481204908402355@4360bbac4327>
physionet-build-dev-1   | 
physionet-build-dev-1   | 
physionet-build-dev-1   | Dear Alistair Edward William Johnson,
physionet-build-dev-1   | 
physionet-build-dev-1   | Tom Pollard has requested to join the Event : Test Event physionet.
physionet-build-dev-1   | 
physionet-build-dev-1   | You can also use the following link to respond to the request:
physionet-build-dev-1   | 
physionet-build-dev-1   | http://localhost:8000/events/
physionet-build-dev-1   | 
physionet-build-dev-1   | 
physionet-build-dev-1   | Regards
physionet-build-dev-1   | The DataShare Team
physionet-build-dev-1   | 
physionet-build-dev-1   | 
physionet-build-dev-1   | -------------------------------------------------------------------------------
```

### CODE REVIEW

File: physionet-django/events/templates/events/email/event_registration.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% load i18n %}{% autoescape off %}{% filter wordwrap:70 %}
+Dear {{ name }},
+
+{{ registered_user_name }} has requested to join the Event : {{ event_title }}
+
+You can also use the following link to respond to the request:
```
Comment (tompollard) [inline physionet-django/events/templates/events/email/event_registration.html]: Remove "also"

File: physionet-django/events/templates/events/email/event_registration.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{% load i18n %}{% autoescape off %}{% filter wordwrap:70 %}
+Dear {{ name }},
+
+{{ registered_user_name }} has requested to join the Event : {{ event_title }}
```
Comment (tompollard) [inline physionet-django/events/templates/events/email/event_registration.html]: change to: 

```
{{ registered_user_name }} has requested to join: {{ event_title }}
```

File: physionet-django/events/templates/events/email/event_registration.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+You can also use the following link to respond to the request:
+
```
Comment (tompollard) [inline physionet-django/events/templates/events/email/event_registration.html:5]: Remove empty line before URL.

File: physionet-django/events/templates/events/email/event_registration.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+{{ url_prefix }}{{ events_home }}
+
+
```
Comment (tompollard) [inline physionet-django/events/templates/events/email/event_registration.html]: Do other emails have two lines before "Regards" or one line?

File: physionet-django/events/utility.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import notification.utility as notification
+
+
+def notify_host_cohosts_new_registration(request, registered_user, event):
+    """
+    Notify the host and cohosts of an event of a message.
```
Comment (tompollard) [inline physionet-django/events/utility.py]: Change to something like:

```
Notify the host and cohosts that a user has submitted a request to join their event.
```

**Review by tompollard — APPROVED** [review]
Thanks Amit, this looks okay to me. I have suggested a couple of minor tweaks to the wording of the email.

**Comment by superryeti** [comment]
> Thanks Amit, this looks okay to me. I have suggested a couple of minor tweaks to the wording of the email.

Thanks @tompollard. Working on these

File: physionet-django/events/templates/events/email/event_registration.html (hunk 4 above)
Comment (superryeti) [reply physionet-django/events/templates/events/email/event_registration.html]: This is interesting. it looks like on `events` app emails we have two lines and on `user` app emails we have single line before `Regards`.

I will add single line for the `event_registration.html` and open another PR for other email templates in events.(unless you recommend i add  one more commit for that here)

File: physionet-django/events/utility.py (hunk 5 above)
Comment (superryeti) [reply physionet-django/events/utility.py]: Ah yes my bad. Originally i was thinking of making this a generic function, but forgot to update the doc strings.

Thanks

File: physionet-django/events/templates/events/email/event_registration.html (hunk 4 above)
Comment (superryeti) [reply physionet-django/events/templates/events/email/event_registration.html]: opened a PR here for extra line https://github.com/MIT-LCP/physionet-build/pull/1897/files

**Comment by superryeti** [comment]
Thanks @tompollard . Applied your suggestions

**Comment by superryeti** [comment]
@tompollard  I think this PR is good to go. 
Although it adds the function which will send emails to cohosts and hosts when someone joins the event; it shouldn't break anything as Benjamin only reverted the code which let us add someone as cohost. Since currently we cant make someone cohosts so no emails will be sent to cohosts.
Once i add the PR to add someone as cohost, the current change should just start working

**Comment by tompollard** [comment]
Thanks Amit, looks good to me/

**Pull request merged**